jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/internal/HttpReceiverOverHTTP.java

```diff
@@ -458,7 +458,7 @@ public boolean content(ByteBuffer buffer)
 
         if (LOG.isDebugEnabled())
             LOG.debug("Setting action to responseContentAvailable on {}", this);
-        if (getAndSetAction(this::responseContentAvailable) != null)
+        if (getAndSetAction(() -> responseContentAvailable(exchange)) != null)
```

Comment:
We are now capturing exchange here, to call a method that itself creates a lambda that capture exchange.    I doubt the JIT will optimize out that double lambda... so feels like it should be able to be done in one with a different API?  Maybe factor out the `responseContentAvailable` method into a `ResponseContentAvailableRunner` class and create an instance of it here?

Reply:
https://github.com/jetty/jetty.project/pull/12203 removed this action mechanism as it uses a different mechanism to avoid re-entering the parser. As a side effect, it doesn't need an allocating lambda anymore.